src/jit/codegencommon.cpp

```diff
@@ -8961,24 +8961,40 @@ void CodeGen::genFnEpilog(BasicBlock* block)
                 NO_WAY("Unsupported JMP indirection");
             }
 
-            const emitter::EmitCallType callType =
-                (addrInfo.accessType == IAT_VALUE) ? emitter::EC_FUNC_TOKEN : emitter::EC_FUNC_TOKEN_INDIR;
+            // If we have IAT_PVALUE, jump via register indirect, as sometimes the
+            // indirection cell can't be reached by the jump.
 
-            // Simply emit a jump to the methodHnd. This is similar to a call so we can use
-            // the same descriptor with some minor adjustments.
+            emitter::EmitCallType callType;
+            void*                 addr;
+            regNumber             indCallReg;
+
+            if (addrInfo.accessType == IAT_PVALUE)
```

Comment:
I'm not terribly familiar with this code, but this seems to be assuming that `addrInfo.accessType` is either `IAT_VALUE` or `IAT_PVALUE`. Should that be asserted?

Reply:
We assert for that at line 8959, just above.